nixos/modules/system/activation/activation-script.nix

```diff
@@ -115,6 +115,49 @@ in
       };
     };
 
+    system.dryActivationScripts = mkOption {
+      default = {};
+      description = ''
+        A set of shell script fragments that are executed when a NixOS
+        system configuration is dry-activated.  Examples are updating
+        /etc, creating accounts, and so on.  These will not be executed
+        on regular system configuration changes but rather at dry system
+        configuration changes.  Since this is run in dry-activate mode,
+        it's important that they don't modify anything about the system.
+      '';
+
+      type = types.attrsOf scriptType;
+
+      apply = set: {
+        script = out:
```

Comment:
in the original `system.activationScripts` this function could be moved out and take the also a `set` parameter.
Then this function could be applied twice for `script` and `dryScript`, where the latter one would receive the filtered set.